src/Sarif.UnitTests/Core/FileDataTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Text;
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class FileDataTests
+    {
+        [Fact]
+        public void FileData_Create_NullUri()
```

Comment:
>FileData_Create_NullUri [](start = 20, length = 23)

I want to advocate for a naming convention where a test name states what it tests. Then the list of tests reads like a specification. (That's exactly what BDD does: you write a list of tests that together constitute a specification).

So for example:
```
FileData_Create_WithNullUri_ThrowsArgumentNullException
```

Reply:
ok. you win. let's update a standards doc somewhere and scrub later.

---
In reply to: [136156139](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136156139) [](ancestors = 136156139)